app/components/istag-select/istag-select.component.js

```diff
@@ -0,0 +1,151 @@
+"use strict";
+
+(function() {
+  angular.module("openshiftConsole").component("istagSelect", {
+  /**
+   * Widget for selecting image stream tags.
+   *
+   * model:
+   *   The model for the input. The model will either use or add the following keys:
+   *     {
+   *       namespace: "",
+   *       imageStream: "",
+   *       tagObject: {
+   *          tag: "",
+   *          items: []
+   *        }
+   *     }
+   *   The tagObject.items array is only present in images that exist and are not pre-populated
+   *
+   * selectDisabled:
+   *   An expression that will disable the form (default: false)
+   */
+    controller: [
+      '$scope',
+      'APIService',
+      'DataService',
+      'ProjectsService',
+      IstagSelect
+    ],
+    controllerAs: '$ctrl',
+    bindings: {
+      istag: '=model',
+      selectDisabled: '=',
+      selectRequired: '=',
+      includeSharedNamespace: '=',
+      allowCustomTag: '=',
+      appendToBody: '='
```

Comment:
Most of these should be one-way bindings with `<` (although `=model` needs to be a two-way binding as you have it).